packages/react/src/components/ContainedList/ContainedList.tsx

```diff
@@ -199,4 +204,6 @@ ContainedList.propTypes = {
   size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
 };
 
+ContainedList.ContainedListItem = ContainedListItem;
```

Comment:
This threw me for a loop, I thought we had deprecated all instances where we were placing subcomponents as fields on the parent component object. Turns out this has been in there since the [inception](https://github.com/carbon-design-system/carbon/pull/11969/files#diff-0de97c9a4278f11b57d633ec584161b202c9123ba4607eddacf559a500bb7f5cR11) of this component.

It's not something we need to do in this PR, but I think it would make sense to deprecate this like we did for TextInput.PasswordInput [here](https://github.com/carbon-design-system/carbon/blob/4943209519239d9021c0cb80cd3d02df2eaeac77/packages/react/src/components/TextInput/index.ts#L13-L20)

```js
// what's currently possible, a bit out of convention
<ContainedList>
  <ContainedList.ContainedListItem/>
  <ContainedList.ContainedListItem/>
</ContainedList>

// following the general convention we have would mean 
// it wouldn't be a field on the object. The above 
// wouldn't work, instead:
<ContainedList>
  <ContainedListItem/>
  <ContainedListItem/>
</ContainedList>
```

I can open a new issue for this 👍

Reply:
https://github.com/carbon-design-system/carbon/issues/16354